packages/compass-e2e-tests/tests/connection.test.ts

```diff
@@ -299,6 +300,29 @@ describe('Connection string', function () {
     }
   });
 
+  it('fails for authentication errors', async function () {
```

Comment:
We didn't actually have any tests for failed connections except in auto-connect.test.ts for connecting from the CLI, so we didn't notice before that the toast never showed up.